docs/concepts/what-is-a-namespace.md

```diff
@@ -10,23 +10,54 @@ tags:
-  - If you are deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace by using tctl; for example, `tctl namespace default`.
-    We recommend using the default Namespace if you aren’t using multiple Namespaces.
+  See the [Registration](#registration) section for details.
 - **Case Insensitive**: Because of DNS, Namespaces are case insensitive on the network and routing side.
-  We recommend using lowercase for namespace names to avoid potential issues.
+  We recommend using lowercase for Namespace names to avoid potential issues.
 - **Membership**: [Task Queue](/concepts/what-is-a-task-queue) names and [Workflow Ids](/concepts/what-is-a-workflow-id) must all correspond to a specific Namespace.
   For example, when a Workflow Execution is spawned, it does so within a specific Namespace.
 - **Uniqueness**: Temporal guarantees a unique Workflow Id within a Namespace.
   Workflow Executions may have the same Workflow Id if they are in different Namespaces.
-- **Namespace Configuration**: Various configuration options like the retention period and the [Archival](/concepts/what-is-archival) destination are configured per Namespace through a special CRUD API or through [`tctl`](/tctl).
+- **Namespace Configuration**: Various configuration options like the [Retention Period](/concepts/what-is-a-retention-period) and the [Archival](/concepts/what-is-archival) destination are configured per Namespace through a special CRUD API or through [`tctl`](/tctl/namespace).
+
+### Registration
+
+Registering a Namespace creates the Namespace on the Temporal Cluster.
+When you register your Namespace, you must also set the [Retention Period](/concepts/what-is-a-retention-period) for the Namespace.
+
+All SDKs require a Namespace on the Cluster for their Client calls. If not set using Client options, the Workflow Client API looks for the `default` Namespace. If there is no default Namespace registered with your Temporal Cluster, all calls will throw errors.
+You must register your Namespace with the Temporal Cluster before setting it in your Client.
+
+You can register your Namespaces in the following ways:
+
+- In your Cluster setup, create your Namespaces, including the default, in your setup script.
+  For example:
+
+  - If deploying through Docker Compose or using the [auto-setup image](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) in a custom Docker Compose application, the Namespace "default" is created, through the auto-setup script.
+  - If deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace by using tctl; for example, `tctl namespace default`.
```

Comment:
I don't think `tctl namespace default` will create you a namespace. Please try it to verify.

Reply:
corrected.